modulefiles/obsproc_common.lua

```diff
@@ -2,17 +2,19 @@ help([[
 Load common modules to build obsproc on all machines
 ]])
 
-hdf5_ver=os.getenv("hdf5_ver") or "default"
-netcdf_ver=os.getenv("netcdf_ver") or "default"
-bacio_ver=os.getenv("bacio_ver") or "default"
-w3emc_ver=os.getenv("w3emc_ver") or "default"
-sp_ver=os.getenv("sp_ver") or "default"
-sigio_ver=os.getenv("sigio_ver") or "default"
-nemsio_ver=os.getenv("nemsio_ver") or "default"
-bufr_ver=os.getenv("bufr_ver") or "default"
+hdf5_ver=os.getenv("hdf5_ver")
```

Comment:
If `hdf5_ver` is not in the environment, then `hdf5_ver` will be set to an empty string in this module file.  This will result in the default module being loaded below, which may or may not be the one you want to use.  If this is not desired, then the only way I have found for this to always raise an error is to provide a non-existent version if `hdf5_ver` is not in the environment, e.g.

```lua
hdf5_ver=os.getenv("hdf5_ver") or "None"
```
This will cause a failure when attempting to load the `hdf5/None` module.

Reply:
Thanks for this info @DavidHuber-NOAA ! Question, would this also be a concern for the other module versions or just specifically `hdf5`?

Reply:
It would be a concern for all versions, I just used `hdf5` as an example.

Reply:
Per your suggestion, I have added `or "None"` to the modulefiles when obtaining the version variables from the environment. Committed @ b6cbdf1